src/dune_rules/module_compilation.ml

```diff
@@ -69,8 +69,20 @@ let build_cm cctx ~precompiled_cmi ~cm_kind (m : Module.t)
       Sandbox_config.needs_sandboxing
     | _ -> default
   in
+  let open Memo.O in
+  let* compiler =
+    let+ compiler =
+      match mode with
+      | Ocaml mode -> Memo.return @@ Context.compiler ctx mode
+      | Melange ->
+        (* TODO loc should come from the mode field in the dune file *)
+        Super_context.resolve_program sctx ~loc:None ~dir
```

Comment:
I wonder about perf implications of this change? I understand as it is using `Memo` it causes no harm having it on such hot path as `build_cm`?

Reply:
It's not something to be concerned about. This code path already contains hundreds of memoized calls. Note that Context.t does things differently because it's an ancient part of the codebase that existed pre memoization. Not necessarily because it's the best way to do things.